avocado/core/test.py

```diff
@@ -276,12 +276,15 @@ def record_and_warn(*args, **kwargs):
             self.__base_logdir_tmp = tempfile.TemporaryDirectory(prefix=prefix)
             self.__base_logdir = self.__base_logdir_tmp.name
 
-        self.__base_logdir = os.path.join(self.__base_logdir, 'test-results')
-        logdir = os.path.join(self.__base_logdir, self.name.str_filesystem)
-        if os.path.exists(logdir):
-            raise exceptions.TestSetupFail("Log dir already exists, this "
-                                           "should never happen: %s"
-                                           % logdir)
+        if self._config.get("run.test_runner") == 'nrunner':
+            logdir = self.__base_logdir
+        else:
+            self.__base_logdir = os.path.join(self.__base_logdir, 'test-results')
+            logdir = os.path.join(self.__base_logdir, self.name.str_filesystem)
+            if os.path.exists(logdir):
+                raise exceptions.TestSetupFail("Log dir already exists, this "
+                                               "should never happen: %s"
+                                               % logdir)
```

Comment:
@clebergnu do we really need implementing legacy runner code at this point?

Reply:
No, but I don't think this is limited to the legacy runner.  I haven't looked at this case in detail, but keep in mind that the `avocado-runner-avocado-instrumented` runner loads and executes a `avocado/core/test.py:Test`.

Reply:
My comment maybe it was not showing properly on GH web interface. This change starts with a if on line 279 that checks `if self._config.get("run.test_runner") == 'nrunner':`. IIUC, this if/else is not needed, and we need only the line 280.

Reply:
I can remove the legacy code from Test class, but I think that is not related to this PR. So I decided to use `if self._config.get("run.test_runner") == 'nrunner':`

Reply:
But this `if` is being introduced here, with your PR. And my point is that we should try to avoid adding legacy code.

Reply:
I still not understand, I am not adding legacy code, I am just avoiding its removal. Because if I remove this `if` statement, I will have to remove whole legacy code from `Test` class. which will breaks the legacy runner itself. So I will have to remove the whole legacy code from avocado because it is strange to me to have some legacy code which is broken inside avocado. Such operation is a huge refactoring, and I don't think that the removal of the whole legacy runner should be part of this PR. That is why I decided to use the `if` statement.

Reply:
IMO, the legacy code don't need to be working at this stage of the master branch. According to @clebergnu at https://github.com/avocado-framework/avocado/issues/4848#issuecomment-989156272:

> runner is being "dismantled" and will go away very soon (it's actually past due)

Unless I'm missing something, we are in "dismantled" mode. And because of that, I'm trying to understand why is a "huge" refactoring. With regarding your patch, we could do something like this:


```diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index e18f9019..4030c855 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -276,13 +276,7 @@ class Test(unittest.TestCase, TestData):
             self.__base_logdir_tmp = tempfile.TemporaryDirectory(prefix=prefix)
             self.__base_logdir = self.__base_logdir_tmp.name
 
-        self.__base_logdir = os.path.join(self.__base_logdir, 'test-results')
-        logdir = os.path.join(self.__base_logdir, self.name.str_filesystem)
-        if os.path.exists(logdir):
-            raise exceptions.TestSetupFail("Log dir already exists, this "
-                                           "should never happen: %s"
-                                           % logdir)
-        self.__logdir = utils_path.init_dir(logdir)
+        self.__logdir = self.__base_logdir
         self.__logfile = os.path.join(self.logdir, 'debug.log')
 
         self._stdout_file = os.path.join(self.logdir, 'stdout')
```
In this way, we are just caring about the 'nrunner'.

But, I might be wrong, that is the reason I pinged @clebergnu.